src/server/node_func.c

```diff
@@ -533,7 +533,8 @@ initialize_pbsnode(struct pbsnode *pnode, char *pname, int ntype)
 	/* or  ATR_DFLAG_ANASSN set in the resource definition          */
 
 	for (prd = svr_resc_def; prd; prd = prd->rs_next) {
-		if (prd->rs_flags & (ATR_DFLAG_FNASSN |  ATR_DFLAG_ANASSN)) {
+		if ((prd->rs_flags & (ATR_DFLAG_FNASSN | ATR_DFLAG_ANASSN)) &&
+									(prd->rs_flags & ATR_DFLAG_MOM)) {
```

Comment:
If you create a consumable resource and create a vnode/node, this resource will appear on the node resources_assigned part even though this resource is never been set on that particular node. The code above is going through all the resources available in `svr_resc_def` and adds them nodes resources_available section. This is not correct.
But there are some consumable resources in the master definition file which is required to be consumed by mom tagged with `ATR_DFLAG_MOM` and has to be added to newly created nodes/vnodes anyways.

Reply:
Along with `ATR_DFLAG_MOM` we are checking for either `ATR_DFLAG_FNASSN` or `ATR_DFLAG_ANASSN`. I think the purpose is to convey the usage in stat command. Between it's set for the following resources on a fresh node - `hbmem, mem, ncpus, vmem`

Reply:
Thanks for the info about  "resource_unset_infinite". I have tested this part. If there is any change in the resources_assigned it will get set in node. So we are safe here.